src/index.rs

```diff
@@ -1133,4 +1106,226 @@ mod tests {
+            }
+            Err(_) => println!("rnd_num error"),
+        }
+
+        return rand_num;
+    }
+
+    fn init_tmp_directory(root_path: PathBuf) {
+        for i in 0..FILE_COUNT {
+            let data_size: u64 = (i + 10).try_into().unwrap();
+            let mut create_file_name = String::from(FILE_NAME);
+            create_file_name.push_str(&i.to_string());
+
+            create_file_at(
+                root_path.clone(),
+                Some(data_size),
+                Some(&create_file_name),
+            );
+        }
+    }
+
+    #[test]
+    fn update_all_compare_track_one_methods() {
+        run_test_and_clean_up(|path| {
+            // Initial folders and files are created in this function.
+            init_tmp_directory(path.clone());
+            let initial_index = ResourceIndex::build(path.clone());
+            let mut index1 = initial_index.clone();
+            let mut index2 = initial_index.clone();
+
+            for i in 0..FILE_COUNT {
+                let mut file_name = String::from(FILE_NAME);
+                file_name.push_str(&i.to_string());
+
+                let mut file_path = path.clone();
+                let mut new_file_path = path.clone();
+                file_path.push(file_name.clone());
+
+                let rand_new_filename: String = new_rand_filename();
+                new_file_path.push(rand_new_filename.clone());
+
+                let old_id: ResourceId;
+                match CanonicalPathBuf::canonicalize(&file_path) {
+                    Ok(canonicalized_path) => {
+                        old_id = initial_index.path2id
+                            [&canonicalized_path.clone()]
+                            .id;
+                    }
+                    Err(_) => {
+                        log::warn!(
+                            "File {} not found",
+                            file_path.to_str().unwrap_or("no_path_buf")
+                        );
+                        continue;
+                    }
+                }
+
+                let update_state = generate_random_update(
+                    path.clone(),
+                    Some(&file_name),
+                    Some(&rand_new_filename),
+                );
+
+                match update_state.try_into() {
+                    Ok(FileAction::CREATE) => {
+                        let _ = index1.track_addition(&new_file_path);
+                    }
+                    Ok(FileAction::UPDATE) => {
+                        let _ = index1.track_update(&file_path, old_id);
+                    }
+                    Ok(FileAction::DELETE) => {
+                        let _ = index1.track_deletion(old_id);
+                    }
+                    Ok(FileAction::MOVE) => {
+                        let _ = index1.track_deletion(old_id);
+                        let _ = index1.track_addition(&new_file_path);
+                    }
+                    Err(_) => println!("rnd_num error"),
+                }
```

Comment:
Are we intentionally ignoring errors here for testing purposes?


If not, we should panic with errors

```suggestion
               match update_state.try_into() {
                    Ok(FileAction::CREATE) => {
                        index1
                            .track_addition(&new_file_path)
                            .expect("Should track addition correctly");
                    }
                    Ok(FileAction::UPDATE) => {
                        index1
                            .track_update(&file_path, old_id)
                            .expect("Should track update correctly");
                    }
                    Ok(FileAction::DELETE) => {
                        index1
                            .track_deletion(old_id)
                            .expect("Should track deletion correctly");
                    }
                    Ok(FileAction::MOVE) => {
                        index1
                            .track_deletion(old_id)
                            .expect("Should track deletion correctly");
                        index1
                            .track_addition(&new_file_path)
                            .expect("Should track addition correctly");
                    }
                    Err(_) => println!("rnd_num error"),
                }
```

Reply:
@tareknaser yes, we must panic and not ignore the errors. But maybe simple unwrapping would be fine in test code?